pkg/skaffold/build/cluster/cluster.go

```diff
@@ -42,11 +43,32 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, tags tag.ImageTags,
 		defer teardownDockerConfigSecret()
 	}
 
-	return build.InParallel(ctx, out, tags, artifacts, b.buildArtifactWithKaniko)
+	// If we only have kaniko builds running in cluster, then we can do the builds in parallel
+	onlyKanikoBuilds := true
+	for _, a := range artifacts {
+		onlyKanikoBuilds = onlyKanikoBuilds && a.ArtifactType.KanikoArtifact != nil
+	}
+	if onlyKanikoBuilds {
+		return build.InParallel(ctx, out, tags, artifacts, b.buildArtifactWithKaniko)
+	}
+
+	// Otherwise, run the builds in sequence.
+	return build.InSequence(ctx, out, tags, artifacts, b.runBuildForArtifact)
```

Comment:
hmm...it's a bit more complicated but I think we should collect the kaniko builds and run them in parallel with building all the custom artifacts in sequence. Building all kaniko artifacts in sequence sounds like a very slow solution and a waste of resources on a cluster.

Reply:
Something we should consider documenting when we add CustomBuilder.

Reply:
With the way this is organized right now, i dont think this is going to used until we support another builder type for in cluster.
Maybe we shd remove it for now? It will help us get our test coverage up too since this branch isn't tested

Reply:
>Is it because we don't know what the scripts do and they might use same resources and could not run in parallel?

Yah, exactly. There's a section in my DD about letting users define if they want their builds to run in parallel or not, which could be an option.

Am happy to remove it for now and add it back later.